### PR TITLE
Implement shard's endpoints used to coordinate creation of a distributed backup

### PR DESCRIPTION
**Pull request by redouan-rhazouani**
This is the second PR of a series of PRs to implement distributed backup operations

### Review checklist

- [x] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/backup/backupper.go [hunk 1]
```diff
@@ -165,3 +166,101 @@ func (b *backupper) objectStore(backend string) (objectStore, error) {
...
+}
+
+// OnCommit will be triggered when the coordinator confirms the execution of a previous operation
+func (b *backupper) OnCommit(ctx context.Context, req *StatusRequest) error {
+	st := b.lastBackup.get()
+	if st.ID == req.ID && b.waitingForCoodinatorToCommit.Load() {
+		b.coordChan <- *req
+		return nil
+	}
+	return fmt.Errorf("shard has abandon backup opeartion")
```
Comment (parkerduckworth) [inline usecases/backup/backupper.go:255]: `abandon` -> `abandoned`

File: usecases/backup/handler.go [hunk 2]
```diff
@@ -240,6 +240,48 @@ func (m *Manager) RestorationStatus(ctx context.Context, principal *models.Princ
...
+// in a distributed backup operation
+func (m *Manager) OnCanCommit(ctx context.Context, req *Request) CanCommitResponse {
+	ret := CanCommitResponse{Method: req.Method, ID: req.ID}
+	store, err := backend(m.backends, req.Backend)
+	if err != nil {
+		ret.Err = fmt.Sprintf("no backup backend %q, did you enable the right module?", req.Backend)
+		return ret
+	}
+
+	// TODO: validate if all req.Classes exist locally
```
Comment (parkerduckworth) [inline usecases/backup/handler.go:253]: What is the current behavior if at least one in `req.Classes` does not exist on the node?

File: usecases/backup/transport.go [hunk 3]
```diff
@@ -49,7 +50,9 @@ type CanCommitResponse struct {
 	// ID is the backup ID
 	ID string
 	// Timeout for how long the promise might be hold
-	Timeout int
+	Timeout time.Duration
+	//
```
Comment (parkerduckworth) [inline usecases/backup/transport.go:54]: Empty comment

File: usecases/backup/handler.go (hunk 2 above)
Comment (redouan-rhazouani) [reply usecases/backup/handler.go:253]: The current behaviour is return an error if any classes does't exist.

File: usecases/backup/backupper.go (hunk 1 above)
Comment (redouan-rhazouani) [reply usecases/backup/backupper.go]: Done 👍

File: usecases/backup/transport.go (hunk 3 above)
Comment (redouan-rhazouani) [reply usecases/backup/transport.go:54]: Will remove it in the next PR 👍

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**